Renaming `github_username` to `github_owner` + generating `__repo_name` & `__repo_url`

matt-graham opened the pull request:
Though its not exposed to user now we have longer prompts rather than using variable names directly, I think `github_username` is a bit of a misnomer given that it can be either a user or organization name, hence I think `github_owner` would be a better choice.

We also currently manually construct the GitHub repository URL in lots of different bits of the template and also the qualified repository name in a few places. We can use [double underscore prefixed (rendered) private variables](https://cookiecutter.readthedocs.io/en/stable/advanced/private_variables.html) in the cookiecutter config to generate the repository URL and qualified name once and then reuse elsewhere to avoid the repetition. This would also make it simpler to later switch to supporting alternative repository hosting options such as GitLab in future (which is the rational for naming the variables `__repo_name` and `__repo_url` rather than `__github_repo_name` and `__github_repo_url`).

matt-graham committed:
Renaming github_user to github_owner + generating repo_name and repo_url

dstansby commented:
I'm 👍 , but reluctant to approve before we have some regression tests (https://github.com/UCL-ARC/python-tooling/issues/329) since there's a lot of changes here that could be susceptible to human error (e.g., missing or adding "/" characters from URLs)

paddyroddy reviewed (approved):
Oo I didn't know about the private variables, big fan of this approach.

Are there any other private variables that we could leverage?

samcunliffe commented:
> I'm 👍 , but reluctant to approve before we have some regression tests (https://github.com/UCL-ARC/python-tooling/issues/329) since there's a lot of changes here that could be susceptible to human error (e.g., missing or adding "/" characters from URLs)

Isn't it enough the tests we currently have, all pass? We do do a cookiecut operation in CI, right? Which is working...

https://github.com/UCL-ARC/python-tooling/blob/2477eee282b58b8083e96013cae35843e10cf6b9/tests/test_package_gen.py#L7

samcunliffe reviewed (approved):
🫐 

(Conflicts, obvs)

matt-graham committed:
Merge branch 'main' into mmg/rename-to-github-owner

matt-graham committed:
Merge branch 'main' into mmg/rename-to-github-owner

paddyroddy committed:
Merge branch 'main' into mmg/rename-to-github-owner

paddyroddy committed:
Merge branch 'main' into mmg/rename-to-github-owner

dstansby committed:
Merge branch 'main' into mmg/rename-to-github-owner

paddyroddy committed:
Merge branch 'main' into mmg/rename-to-github-owner

paddyroddy committed:
Merge branch 'main' into mmg/rename-to-github-owner

paddyroddy committed:
Merge branch 'main' into mmg/rename-to-github-owner

matt-graham committed:
Change github_username to github_owner in package generation test

matt-graham committed:
Change references to github_user in tutorial to github_owner

matt-graham commented:
Tests were previously failing since #432 was merged in as the passed configs now need to use `github_owner` rather than `github_username` - now fixed and have also updated tutorial to make more consistent with new `github_owner` variable name.

samcunliffe reviewed (approved):
🫐

samcunliffe commented:
Let's make `v1.0.0` of the template when this is in.

paddyroddy reviewed (approved):
🫐

Saransh-cpp commented on cookiecutter.json:
```diff
@@ -8,11 +8,13 @@
   "max_python_version": ["3.13", "3.12", "3.11"],
   "license": ["MIT", "BSD-3", "GPL-3.0"],
   "funder": "",
+  "__repo_name": "{{cookiecutter.github_owner}}/{{cookiecutter.project_slug}}",
```
TIL `__variable` in `cookiecutter.json`, looks so clean!
